File: gherkin/dart/lib/src/exceptions/NoSuchLanguageException.dart
import 'package:gherkin/language.dart';

import 'ParserException.dart';

class NoSuchLanguageException extends ParserException
{
  NoSuchLanguageException(String language, [Location location=Location.empty])
    : super('Language not supported: $language', location)
  {
    if (language.isEmpty) {
      throw ArgumentError.notNull('language');
    }
  }
}

File: gherkin/php/src/ParserException/NoSuchLanguageException.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin\ParserException;

use Cucumber\Gherkin\Location;
use Cucumber\Gherkin\ParserException;

final class NoSuchLanguageException extends ParserException
{
    public function __construct(string $language, ?Location $location)
    {
        parent::__construct('Language not supported: ' . $language, $location);
    }
}
